### PR TITLE
Remove ifdefs related to formatting

### PR DESCRIPTION
**Pull request by tmat**
Use `ISyntaxFormatting` and inline formatting helpers.

### CODE REVIEW

File: src/Analyzers/Core/Analyzers/Formatting/AbstractFormattingAnalyzer.cs [hunk 1]
```diff
@@ -41,7 +43,63 @@ private void AnalyzeSyntaxTree(SyntaxTreeAnalysisContext context, CompilationOpt
         if (ShouldSkipAnalysis(context, compilationOptions, notification: null))
             return;
 
-        var options = context.GetAnalyzerOptions().GetSyntaxFormattingOptions(SyntaxFormatting);
-        FormattingAnalyzerHelper.AnalyzeSyntaxTree(context, SyntaxFormatting, Descriptor, options);
+        var tree = context.Tree;
```
Comment (tmat) [inline src/Analyzers/Core/Analyzers/Formatting/AbstractFormattingAnalyzer.cs:46]: Inlinded from FormattingAnalyzerHelper

File: src/Analyzers/Core/Analyzers/Formatting/FormatterHelper.cs [hunk 2]
```diff
@@ -1,68 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
```
Comment (tmat) [inline src/Analyzers/Core/Analyzers/Formatting/FormatterHelper.cs:1]: All inlined.

File: src/Analyzers/Core/Analyzers/Formatting/FormattingAnalyzerHelper.cs [hunk 3]
```diff
@@ -1,75 +0,0 @@
...
-using Microsoft.CodeAnalysis.Text;
-using Roslyn.Utilities;
-using Formatter = Microsoft.CodeAnalysis.Formatting.FormatterHelper;
-using FormattingProvider = Microsoft.CodeAnalysis.Formatting.ISyntaxFormatting;
-
-namespace Microsoft.CodeAnalysis.CodeStyle;
-
-internal static class FormattingAnalyzerHelper
-{
-    internal static void AnalyzeSyntaxTree(SyntaxTreeAnalysisContext context, FormattingProvider formattingProvider, DiagnosticDescriptor descriptor, SyntaxFormattingOptions options)
```
Comment (tmat) [inline src/Analyzers/Core/Analyzers/Formatting/FormattingAnalyzerHelper.cs:18]: Inlined to caller.

File: src/Analyzers/Core/CodeFixes/Formatting/FormattingCodeFixHelper.cs [hunk 4]
```diff
@@ -1,32 +0,0 @@
...
-using Microsoft.CodeAnalysis.Text;
-
-using Formatter = Microsoft.CodeAnalysis.Formatting.FormatterHelper;
-using FormattingProvider = Microsoft.CodeAnalysis.Formatting.ISyntaxFormatting;
-
-namespace Microsoft.CodeAnalysis;
-
-internal static class FormattingCodeFixHelper
-{
-    internal static async Task<SyntaxTree> FixOneAsync(SyntaxTree syntaxTree, FormattingProvider formattingProvider, SyntaxFormattingOptions options, Diagnostic diagnostic, CancellationToken cancellationToken)
```
Comment (tmat) [inline src/Analyzers/Core/CodeFixes/Formatting/FormattingCodeFixHelper.cs:17]: Inlined and simplified.

File: src/Analyzers/VisualBasic/CodeFixes/RemoveUnusedParametersAndValues/VisualBasicRemoveUnusedValuesCodeFixProvider.vb [hunk 5]
```diff
@@ -25,9 +25,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.RemoveUnusedParametersAndValues
...
         End Sub
 
-        Protected Overrides Function GetSyntaxFormatting() As ISyntaxFormatting
-            Return VisualBasicSyntaxFormatting.Instance
-        End Function
+        Protected Overrides ReadOnly Property SyntaxFormatting As ISyntaxFormatting
+            Get
+                Return VisualBasicSyntaxFormatting.Instance
+            End Get
+        End Property
```
Comment (CyrusNajmabadi) [inline src/Analyzers/VisualBasic/CodeFixes/RemoveUnusedParametersAndValues/VisualBasicRemoveUnusedValuesCodeFixProvider.vb:32]:
```suggestion
        Protected Overrides ReadOnly Property SyntaxFormatting As ISyntaxFormatting = VisualBasicSyntaxFormatting.Instance
```

File: src/Analyzers/VisualBasic/CodeFixes/UseConditionalExpression/VisualBasicUseConditionalExpressionForAssignmentCodeFixProvider.vb [hunk 6]
```diff
@@ -56,8 +56,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.UseConditionalExpression
...
         End Function
 
-        Protected Overrides Function GetSyntaxFormatting() As ISyntaxFormatting
-            Return VisualBasicSyntaxFormatting.Instance
-        End Function
+        Protected Overrides ReadOnly Property SyntaxFormatting As ISyntaxFormatting
+            Get
+                Return VisualBasicSyntaxFormatting.Instance
+            End Get
+        End Property
```
Comment (CyrusNajmabadi) [inline src/Analyzers/VisualBasic/CodeFixes/UseConditionalExpression/VisualBasicUseConditionalExpressionForAssignmentCodeFixProvider.vb:63]:
```suggestion
        Protected Overrides ReadOnly Property SyntaxFormatting As ISyntaxFormatting = VisualBasicSyntaxFormatting.Instance
```

File: src/Analyzers/VisualBasic/CodeFixes/UseConditionalExpression/VisualBasicUseConditionalExpressionForReturnCodeFixProvider.vb [hunk 7]
```diff
@@ -40,8 +40,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.UseConditionalExpression
...
         End Function
 
-        Protected Overrides Function GetSyntaxFormatting() As ISyntaxFormatting
-            Return VisualBasicSyntaxFormatting.Instance
-        End Function
+        Protected Overrides ReadOnly Property SyntaxFormatting As ISyntaxFormatting
+            Get
+                Return VisualBasicSyntaxFormatting.Instance
+            End Get
+        End Property
```
Comment (CyrusNajmabadi) [inline src/Analyzers/VisualBasic/CodeFixes/UseConditionalExpression/VisualBasicUseConditionalExpressionForReturnCodeFixProvider.vb:47]:
```suggestion
        Protected Overrides ReadOnly Property SyntaxFormatting As ISyntaxFormatting = VisualBasicSyntaxFormatting.Instance
```

**Review by CyrusNajmabadi — APPROVED** [review]

**Pull request merged**